app/components/learningmaterial-search.js

```diff
@@ -1,99 +1,101 @@
 import Ember from 'ember';
 
-const { Component, computed, inject, run } = Ember;
+const { Component, inject } = Ember;
 const { service } = inject;
-const { debounce } = run;
-const { notEmpty, or } = computed;
 
-export default Component.extend({
-  classNames: ['learningmaterial-search'],
+var ProxiedMaterials = Ember.ObjectProxy.extend({
+  isActive: false
+});
 
+export default Component.extend({
   store: service(),
   i18n: service(),
-
-  results: [],
+  classNames: ['learningmaterial-search'],
   currentMaterials: [],
+  query: '',
+  searchResults: [],
+  searchPage: 0,
+  searchResultsPerPage: 50,
+  hasMoreSearchResults: false,
+  targetItemTitle: '',
   searching: false,
-  showMoreInputPrompt: false,
-  showNoResultsMessage: false,
-  currentlySearchingForTerm: false,
-
-  hasResults: notEmpty('results'),
-
-  showList: or('searching', 'showMoreInputPrompt', 'showNoResultsMessage', 'hasResults'),
+  searchReturned: false,
 
   actions: {
-    clear() {
-      let input = this.$('input')[0];
-      input.value = '';
-
-      this.setProperties({
-        searchTerms: '',
-        showMoreInputPrompt: false,
-        showNoResultsMessage: false,
-        searching: false,
-        results: [],
-        showClearButton: false,
-      });
-    },
-
-    inputValueChanged() {
-      let input = this.$('input')[0];
-      let searchTerms = input.value;
-
-      if (this.get('currentlySearchingForTerm') === searchTerms) {
+    search: function(query){
+      var self = this;
+      if (Ember.$.trim(query) === '') {
+        self.set('searchReturned', false);
+        self.set('searching', false);
+        self.set('searchPage', 1);
+        self.set('hasMoreSearchResults', false);
+        self.set('searchResults', []);
         return;
       }
-
-      this.setProperties({
-        currentlySearchingForTerm: searchTerms,
-        showMoreInputPrompt: false,
-        showNoResultsMessage: false,
-        searching: false,
-      });
-
-      let noWhiteSpaceTerm = searchTerms.replace(/ /g,'');
-
-      if (noWhiteSpaceTerm.length === 0){
-        this.send('clear');
-        return;
-      } else if (noWhiteSpaceTerm.length < 3) {
-        this.setProperties({
-          results: [],
-          showMoreInputPrompt: true,
-        });
-
-        return;
-      }
-
+      this.set('searchReturned', false);
       this.set('searching', true);
-
-      debounce(this, function() {
-        this.send('search', searchTerms);
-      }, 300);
+      this.set('query', query);
+      var currentMaterials = this.get('currentMaterials');
+      this.get('store').query('learningMaterial', {
+        q: query,
+        limit: this.get('searchResultsPerPage') + 1,
+        'order_by[title]': 'ASC',
+      }).then(function(lms){
+        let results = lms.map(function(lm){
+          return ProxiedMaterials.create({
```

Comment:
Instead of doing this as a proxy can we just use a helper in the template?
Like: `{{contains material currentMaterials}}`